### ISSUE TITLE
Add ability to pull just the embed URL or just the video id itself

### ISSUE THREAD

**Issue by TUSAB-web** [wrav/oembed#57]
I have a page with lots of videos, and embedding all of them slows it to a crawl. I found this method to embed them without hurting performance:

https://css-tricks.com/lazy-load-embedded-youtube-videos/

This method requires me to be able to pull the embed URL or even just the video id (that seemingly random string unique to each YouTube video). I don't seem to be able to pull only that information - I get more than I want and I don't have an easy way to strip out the extra.

If something like {{ entry.field.media.videoid }} returned just that string, that could easily be inserted into any kind of unique embed.

Alternately, there is this method, but it also requires the video id:

https://github.com/paulirish/lite-youtube-embed

Finally, perhaps if you wanted to get fancy, you could implement either of the "lite youtube" methods above as an option in the plug-in.

**Comment by TUSAB-web** [comment]
I saw that the plugin Wordsmith can grab the video ID from a YouTube link... I will try and combine that with the output from your plugin and see if it works. I will report back on how it works.

**Comment by terryupton** [comment]
@TUSAB-web did you find a way (not using wordsmith) to get the embed url as I want to lazy-load the iframe/embed as well.

**Comment by TUSAB-web** [comment]
> @TUSAB-web did you find a way (not using wordsmith) to get the embed url as I want to lazy-load the iframe/embed as well.

I ended up using Wordsmith - once to pull the YouTube ID and also to pull the stuff after the question mark in the URL in case the video starts at a specific timestamp. I actually ended up just making my field a regular URL field instead of using oEmbed because I realized I wasn’t using its functionality, as wordsmith was pulling out the video ID for me, and the custom code linked above was pulling the poster image. Let me know if you want to see the code I ended up using; but since it doesn’t use oEmbed, I won’t put it in this thread.
